src_c/surface.c

```diff
@@ -1879,6 +1890,77 @@ surf_blit(pgSurfaceObject *self, PyObject *args, PyObject *keywds)
     return pgRect_New(&dest_rect);
 }
 
+static PyObject *
+surf_rblit(pgSurfaceObject *self, PyObject *const *args, Py_ssize_t nargs)
+{
+    SDL_Surface *src, *dest = pgSurface_AsSurface(self);
+    SDL_Rect dest_rect, unused_rect;
+    SDL_Rect *src_rect;
+    PyObject *argpos;
+    PyObject *srcobject;
+    int result, flags_numeric = 0;
+
+    if (nargs == 0 || nargs > 3)
+        return RAISE(
+            PyExc_ValueError,
+            "Function requires at least 2 positional arguments in the order: "
+            "source, dest, (special_flags optional)");
+    else if (nargs == 3)
+        if (!pg_IntFromObj(args[2], &flags_numeric))
+            return RAISE(PyExc_TypeError,
+                         "special_flags parameter must be numeric");
+    srcobject = args[0];
+    src = pgSurface_AsSurface(srcobject);
+    if (!dest || !src)
+        return RAISE(pgExc_SDLError, "display Surface quit");
+
+    argpos = args[1];
+
+    /* Setup the destination rect, src rect is NULL */
+    if (pg_TwoIntsFromObj(argpos, &(dest_rect.x), &(dest_rect.y))) {
+    }
+    else if ((src_rect = pgRect_FromObject(argpos, &unused_rect))) {
+        dest_rect.x = src_rect->x;
+        dest_rect.y = src_rect->y;
+    }
+    else
+        return RAISE(PyExc_TypeError, "invalid destination position for blit");
+
+    dest_rect.w = src->w;
+    dest_rect.h = src->h;
+
+    result = pgSurface_Blit(self, (pgSurfaceObject *)srcobject, &dest_rect,
+                            NULL, flags_numeric);
+
+    if (result != 0)
+        return RAISE(PyExc_TypeError,
+                     "Blit failed (probably the flag used does not exist)");
+
```

Comment:
This should work like in surf_blit. If pgSurface_Blit failed, it already set the error message, so you just need to return NULL.